lib/mdns/client.ex

```diff
@@ -70,6 +74,14 @@ defmodule Mdns.Client do
     {:reply, state.devices, state}
   end
 
+  def handle_call(:stop, _from, state) do
```

Comment:
can we do the pattern matching in the function head for the udp variable?

something about setting `udp` to nil outside of the if logic bugs me.

something like this

```elixir
def handle_call(:stop, _from, %State{udp: nil} = state), do: {:reply, :ok, state}

def handle_call(:stop, _from, %State{udp: udp} = state) do
  :ok = :gen_udp.close(udp)
  {:reply, :ok, %State{state | udp: nil}}
end
```

Reply:
Sure, I just did it the way it was done in the Server module.
Do you want me to change it there also?

Reply:
that would be great. not sure how that snuck in there :)

Reply:
Done :)